### ISSUE TITLE
Adjust color and font family of the address line 2 link in the Checkout block

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#49482]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

With #47160, we've enabled merchants to define the address line 2 field as optional. If it's defined as optional and there's no pre-existing value available, the address line 2 field is hidden behind a link. That link has no defined color (or font family), rendering the link black. If a merchant is using a dark mode theme, the link is almost impossible to read.

<table>
<tr>
<td valign="top">Dark mode theme with dark mode setting enabled:
<br><br>

![Screenshot 2024-07-12 at 16 49 51](https://github.com/user-attachments/assets/c807437c-ac07-4224-914c-8bed7f955bc7)
</td>
<td valign="top">Dark mode theme with dark mode setting disabled:
<br><br>

![Screenshot 2024-07-12 at 16 50 23](https://github.com/user-attachments/assets/86fd70bd-d92c-464d-a3fb-4d4b6eb2e7c5)
</td>
<td valign="top">Light mode theme with dark mode setting disabled (no issues here):
<br><br>

![Screenshot 2024-07-12 at 16 50 53](https://github.com/user-attachments/assets/82ea06b6-5e59-4675-afb3-141feed9a318)
</td>
</tr>
</table>

### Expected behavior

When using a dark mode theme, independent if the dark mode settings are enabled or not in the Checkout block, the `+ Add apartment, suite, etc.` link **is barely visible**.

### Actual behavior

When using a dark mode theme, independent if the dark mode settings are enabled or not in the Checkout block, the `+ Add apartment, suite, etc.` link **is visible**.

### Steps to reproduce

1. Install and activate the Twenty Twenty-Four theme.
2. Go to `Apearance » Editor » Styles` and select the Onyx style.
3. Open the page with the Checkout block in the editor.
4. Define the address line 2 field as optional.
5. Open an incognito window.
6. Add a product to the cart.
7. Go to the checkout page, which contains the Checkout block.
8. See that the `+ Add apartment, suite, etc.` link **is barely visible**.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.2.0-dev
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.5.5
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.4
PHP Version: 8.3.6
PHP Post Max Size: 512 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.7.1
(SecureTransport) OpenSSL/3.2.1

SUHOSIN Installed: –
MySQL Version: 8.3.0
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 24
custom-css-js: 1
page: 11
post: 16
product: 21
product_variation: 7
revision: 112
shop_coupon: 2
shop_order: 47
wp_global_styles: 4
wp_navigation: 1
wp_template_part: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (8) ###

Query Monitor: by John Blackbourn – 3.16.3
Code Snippets: by Code Snippets Pro – 3.6.5.1
Debug Log Manager: by Bowo – 2.3.3
Email Log: by Sudar – 2.4.9
Loco Translate: by Tim Whitlock – 2.6.10
Say What?: by Ademti Software – 2.2.3
SMNTCS Show Symlinked Plugins: by Niels Lange – 1.3
WooCommerce: by Automattic – 9.2.0-dev

### Inactive Plugins (10) ###

Additional Checkout Fields Tester: by WooCommerce – 1.0
Multiple Packages for WooCommerce: by Erica Dion – 1.1.1
ReplaceCartThumbnail: by  –
Show Current Template: by JOTAKI Taisuke – 0.5.2
Simple Custom CSS and JS: by SilkyPress.com – 3.47
SMNTCS Custom Shipping Method: by Your Name – 1.0
SMNTCS Nord Admin Theme: by Niels Lange – 1.4
SMNTCS Theme List View: by Niels Lange – 1.3
Test template override plugin: by mlaetitia – 0.1.0
WooCommerce Stripe Gateway: by WooCommerce – 8.4.0

### Dropin Plugins () ###

db.php: Query Monitor Database Class (Drop-in)

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
mix and match (mix-and-match)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: ✔

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 124 KB

### WC Pages ###

Shop base: #54 - /products-beta/classic-shop/
Cart: #56 - /cart/ -  Contains the woocommerce/cart block
Checkout: #57 - /checkout/ -  Contains the woocommerce/checkout block
My account: #60 - /my-account/
Terms and conditions: #61 - /terms/

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/error.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/notice.php
/Users/nielslange/Plugins/woocommerce/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
experimental-blocks
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
minified-js
mobile-app-banner
navigation
onboarding
onboarding-tasks
payment-gateway-suggestions
product-custom-fields
remote-inbox-notifications
remote-free-extensions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
async-product-editor-category-field
launch-your-store

Disabled Features: pattern-toolkit-full-composability
product-pre-publish-modal
printful
settings
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-07-13 04:44:14 +00:00
Options: ✔
Notes: 86
Onboarding: -

### Action Scheduler ###

Complete: 382
Oldest: 2024-06-11 16:12:45 +0000
Newest: 2024-07-12 14:51:09 +0000

Failed: 9
Oldest: 2024-03-27 17:33:01 +0000
Newest: 2024-06-28 12:40:47 +0000

Pending: 3
Oldest: 2024-07-12 15:43:26 +0000
Newest: 2024-07-13 09:58:02 +0000


### Status report information ###

Generated at: 2024-07-12 15:22:46 +00:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by Manussakis** [comment]
I have started working on this issue.
